sunpy/net/helioviewer.py

```diff
@@ -22,44 +22,75 @@
 
 
 class HelioviewerClient(object):
-    """Helioviewer.org Client"""
-    def __init__(self, url="https://legacy.helioviewer.org/api/"):
+    """Helioviewer.org Client"""      
+    def __init__(self, url="https://api.helioviewer.org/"):
         """
-        url : location of the Helioviewer API.  The default location points to
-            version 1 of the API.  Version 1 of the Helioviewer API is
-            currently planned to be supported until the end of April 2017.
+        url : str
+            URL that points to the Helioviewer API.
         """
+        self.datasource_info = dict()
         self._api = url
-
+        self.init_src_dict()
+    
+    def init_src_dict(self):
+        datasources = self.get_data_sources()
+        for name, observ in datasources.items():
+            if name == "TRACE":
+                for instr, params in observ.items():
+                    self.datasource_info[(name, None, None, instr)] = params['sourceId']
+            else:
+                for inst, detect in observ.items():
+                    for wavelength, params in detect.items():
+                        if name in ["Hinode", "STEREO_A", "STEREO_B"] or wavelength in ["C2", "C3"]:
+                            for wave, adict in params.items():
+                                self.datasource_info[(name, inst, wavelength, wave)] = adict['sourceId']
+                        else:
+                            self.datasource_info[(name, inst, None, wavelength)] = params['sourceId']
+        
     def get_data_sources(self, **kwargs):
         """
         Returns a structured list of datasources available at helioviewer.org.
+
+        Parameters
+        ----------
+        verbose: `bool`
+            (Optional) If set to True, it outputs the hierarchical list in an
+            alternative format and limits the list of available datasources
+            to a known set (SDO and SOHO).
+        enable: `str`
+            (Optional) Comma-separated list of observatories to enable.
+        callback: `str`
+            (Optional) Wrap the response object in a function call of your choosing.
+
```

Comment:
Just remove this `Parameters` please.

Reply:
The whole parameter section of get_data_source?

Reply:
Ok no problem I will remove it